### PR TITLE
Add more structured output for tests

### LINKED ISSUE #582: Debugging broken QL unit tests is difficult
The VSCode console does not show you where an error is located. The error location is only available in the log file. 

The error location should be printed to the output console.

### PR DESCRIPTION
**Pull request by aeisenberg**
The diff and the errors were always available, but they were not being sent to the output.

Additionally, make sure to send output to both the test explorer log and the codeql test log.

Resolves #582.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/test-adapter.ts
Comment (adityasharad) [inline extensions/ql-vscode/src/test-adapter.ts:227]: Would `failed due to errors` or `failed to run` be more meaningful for the user? I had to think about the difference between 'errored' and 'failed' for a moment.

**Review by adityasharad — COMMENTED** [review]
Should be good to go after a rebase.

File: extensions/ql-vscode/src/test-adapter.ts
Comment (aeisenberg) [reply extensions/ql-vscode/src/test-adapter.ts:223]: That won't work. `state` is a property of the `TestEvent` interface and only allows `'running' | 'passed' | 'failed' | 'skipped' | 'errored'`. The hope is that the messages will be more explicit as to what happened.

Comment (adityasharad) [reply extensions/ql-vscode/src/test-adapter.ts:223]: Oh fair enough. We can consider having more detail in the way `message` is constructed below, but no need to block this PR.

**Review by adityasharad — APPROVED** [review]

**Pull request merged**
